be/src/runtime/result_file_sink.cpp

```diff
@@ -119,6 +116,9 @@ Status ResultFileSink::prepare(RuntimeState* state) {
 
     }
     RETURN_IF_ERROR(_writer->init(state));
+    for (int i = 0; i < _channels.size(); ++i) {
```

Comment:
So this is a bug?

Reply:
Yes, if the _writer failed to init, the channel should not be inited before closed. Otherwise, the be will core in destructor